components/net/fetch/request.rs

```diff
@@ -544,24 +543,23 @@ fn http_fetch(request: Rc<RefCell<Request>>,
 }
 
 /// [HTTP network or cache fetch](https://fetch.spec.whatwg.org#http-network-or-cache-fetch)
-fn http_network_or_cache_fetch(request: Rc<RefCell<Request>>,
+fn http_network_or_cache_fetch(request: Rc<Request>,
                                credentials_flag: bool,
                                authentication_fetch_flag: bool) -> Response {
 
+    // Step 1
+    // should this ever not be done?
+    let http_request = request.clone()
```

Comment:
After reading the spec, what you had before made sense, but not on the `else` branch. It should have been something like `Rc::try_unwrap(request).ok().clone()` instead.